src/main/java/software/amazon/smithy/lsp/SmithyLanguageServer.java

```diff
@@ -378,29 +372,31 @@ public CompletableFuture<List<? extends Location>> selectorCommand(SelectorParam
             return completedFuture(Collections.emptyList());
         }
 
-        Project project = projects.mainProject();
-        // TODO: Might also want to tell user if the model isn't loaded
-        // TODO: Use proper location (source is just a point)
-        return completedFuture(project.modelResult().getResult()
+        // Select from all available projects
+        Collection<Project> detached = projects.detachedProjects().values();
+        Collection<Project> nonDetached = projects.nonDetachedProjects().values();
+
+        return completedFuture(Stream.concat(detached.stream(), nonDetached.stream())
+                .flatMap(project -> project.modelResult().getResult().stream())
                 .map(selector::select)
-                .map(shapes -> shapes.stream()
+                .flatMap(shapes -> shapes.stream()
+                        // TODO: Use proper location (source is just a point)
                         .map(Shape::getSourceLocation)
-                        .map(LspAdapter::toLocation)
-                        .collect(Collectors.toList()))
-                .orElse(Collections.emptyList()));
+                        .map(LspAdapter::toLocation))
+                .toList());
     }
 
     @Override
     public CompletableFuture<ServerStatus> serverStatus() {
-        OpenProject openProject = new OpenProject(
-                LspAdapter.toUri(projects.mainProject().root().toString()),
-                projects.mainProject().smithyFiles().keySet().stream()
-                        .map(LspAdapter::toUri)
-                        .collect(Collectors.toList()),
-                false);
-
         List<OpenProject> openProjects = new ArrayList<>();
-        openProjects.add(openProject);
+        for (Project project : projects.nonDetachedProjects().values()) {
+            openProjects.add(new OpenProject(
+                    LspAdapter.toUri(project.root().toString()),
```

Comment:
nit: if you do the to-string conversion any other places maybe add an overloaded method on the adapter

Reply:
Rare, not using `Path` much